java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,351 @@
+            case TINYINT:
+                return row.getByte(index);
+            case SMALLINT:
+                return row.getShort(index);
+            case INTEGER:
+                return row.getInt(index);
+            case BIGINT:
+                return row.getLong(index);
+            case FLOAT:
+                return row.getFloat(index);
+            case DOUBLE:
+                return row.getDouble(index);
+            default:
+                throw new IOException(
+                        String.format("Unsupported data type: %s", fieldTypes.get(index)));
+        }
+    }
+
+    private double getDoubleValue(ArrayData array, int index, LogicalType type) throws IOException {
+        switch (type.getTypeRoot()) {
+            case DECIMAL:
+                final int precision = ((DecimalType) type).getPrecision();
+                final int scale = ((DecimalType) type).getScale();
+                final DecimalData decimal = array.getDecimal(index, precision, scale);
+                return decimal.toBigDecimal().doubleValue();
+            case TINYINT:
+                return array.getByte(index);
+            case SMALLINT:
+                return array.getShort(index);
+            case INTEGER:
+                return array.getInt(index);
+            case BIGINT:
+                return array.getLong(index);
+            case FLOAT:
+                return array.getFloat(index);
+            case DOUBLE:
+                return array.getDouble(index);
+            default:
+                throw new IOException(
+                        String.format("Unsupported data type: %s", fieldTypes.get(index)));
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        super.close();
+        if (config.get(DELETE_ON_SHUTDOWN)) {
+            pushGateway.delete(jobName);
+        }
+    }
+
+    static Map<String, String> parseLabels(final String labelsConfig) {
+        if (!labelsConfig.isEmpty()) {
+            Map<String, String> labelNames = new HashMap<>();
+            String[] kvs = labelsConfig.split(";");
+            for (String kv : kvs) {
+                int idx = kv.indexOf("=");
+                if (idx < 0) {
+                    throw new RuntimeException(
+                            String.format("Invalid prometheusPushGateway labels:%s", kv));
+                }
+
+                String labelName = kv.substring(0, idx);
+                String labelValue = kv.substring(idx + 1);
+                if (StringUtils.isNullOrWhitespaceOnly(labelName)) {
+                    throw new RuntimeException(
+                            String.format("Invalid labelName:%s must not be empty", labelName));
+                }
+                labelNames.put(labelName, labelValue);
+            }
+
+            return labelNames;
+        }
+
+        return Collections.emptyMap();
+    }
+
+    private static class MetricMeta implements Serializable {
```

Comment:
It seems `metadata` is very rarely abbreviated as `meta` (if any) in Flink or other project. Would it be better to use `MetricMetadata` here?